packages/amqp/lib/AmqpQueuePublisherManager.ts

```diff
@@ -0,0 +1,111 @@
+import type {
+  BaseEventType,
+  CommonEventDefinition,
+  EventRegistry,
+  MetadataFiller,
+} from '@message-queue-toolkit/core'
+import { AbstractPublisherManager } from '@message-queue-toolkit/core'
+import type { MessageMetadataType } from '@message-queue-toolkit/core/lib/messages/baseMessageSchemas'
+import type z from 'zod'
+
+import type { AbstractAmqpPublisher, AMQPPublisherOptions } from './AbstractAmqpPublisher'
+import type {
+  AbstractAmqpQueuePublisher,
+  AmqpQueueMessageOptions,
+} from './AbstractAmqpQueuePublisher'
+import type { AMQPCreationConfig, AMQPDependencies, AMQPLocator } from './AbstractAmqpService'
+import type { AmqpPublisherFactory } from './CommonAmqpPublisherFactory'
+import { CommonAmqpQueuePublisherFactory } from './CommonAmqpPublisherFactory'
+
+export type AmqpAwareEventDefinition = {
+  schemaVersion?: string
+  exchange?: string
+  queueName?: string
+} & CommonEventDefinition
+
+export type AmqpPublisherManagerDependencies<SupportedEvents extends AmqpAwareEventDefinition[]> = {
+  eventRegistry: EventRegistry<SupportedEvents>
+} & AMQPDependencies
+
+export type AmqpPublisherManagerOptions<
+  T extends AbstractAmqpPublisher<EventType, MO>,
+  MO,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  PO extends Omit<AMQPPublisherOptions<any>, 'creationConfig'>,
```

Comment:
Not sure if possible, but just wondering if we could be possible to use unknown instead of any

Reply:
not unknown, but I'll try to wrestle with types to make this unnecessary